app/models/admin_setting.rb

```diff
@@ -51,7 +53,8 @@ def self.guest_downloading_off?
     self.first ? self.first.guest_downloading_off? : false
   end
   def self.default_skin
-    self.first ? (self.first.default_skin_id ? self.first.default_skin : Skin.default) : Skin.default
+    default_skin=Rails.cache.fetch(default_skin_cache_key){Skin.default}
```

Comment:
@zz9pzza: would you obey the Hound's comments with regard to spacing, please? The nested ternary was already there so I guess we can ignore it, but the spacing is really bad.